specification/ai/Azure.AI.Projects/agents/messages/models.tsp

```diff
@@ -1,11 +1,104 @@
 import "../tools/models.tsp";
-
 import "@typespec/versioning";
 
 namespace Azure.AI.Projects.Agents;
 
 using TypeSpec.Versioning;
 
+#suppress "@azure-tools/typespec-autorest/union-unsupported" "We intentionally want a union of string | array."
+@doc("Because OpenAPI 2 doesn’t fully support oneOf for 'string or array,' we suppress the rule.")
+union CreateMessageContent {
+  @doc("If callers only have a quick text message, they can pass a plain string here.")
+  text: string,
+
+  @doc("For 'rich' content, callers can pass an array of typed content blocks (e.g. text, image_file, image_url).")
+  blocks: MessageContentBlockInput[],
+}
+
+@doc("For the block's 'type' property, we define an open union with a fallback string to satisfy the no-string-discriminator rule.")
+union MessageBlockType {
+  @doc("A text content block.")
+  text: "text",
+
+  @doc("An internally uploaded image file.")
```

Comment:
Do you need to say that only one of them (image_file , image_url) can be set? Not both together.

Reply:
I'll clarify this more and assuming this is documentation related and can bring it in the next update.